fetch/main.ts

```diff
@@ -0,0 +1,21 @@
+interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+async function getUserArray(): Promise<void | User[]> {
+  try {
+    const response = await fetch('https://jsonplaceholder.typicode.com/users');
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    const formattedResponse = (await response.json()) as Promise<User[]>;
```

Comment:
Since we're using type assertion here we can just type this as `User[]` since we are just telling TypeScript what the end result is.